client/src/components/Atoms/Steps.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Card, Colors, H5, Icon } from '@blueprintjs/core'
+import { assert } from '../utilities'
+
+/**
+ * A set of components to display a multi-step process. Consists of a card
+ * containing:
+ *  - a step progress list (top)
+ *  - a step content panel (middle)
+ *  - a step actions bar (bottom)
+ *
+ * These components are mostly graphical - the parent must manage relevant state
+ * (current step, navigation, disabling, etc.)
+ *
+ * Example usage:
+ *
+ *  <Steps>
+ *    <StepList>
+ *      <StepListItem>Log In</StepListItem>
+ *      <StepListItem current>Prepare</StepListItem>
+ *      <StepListItem>Audit Ballots</StepListItem>
+ *    </StepList>
+ *    <StepPanel>Prepare your ballots</StepPanel>
+ *    <StepActions
+ *      left={<Button>Back</Button>}
+ *      right={<Button>Next</Button>}
+ *    />
+ *  </Steps>
+ */
+
+export const Steps = styled(Card).attrs({ elevation: 1 })`
+  padding: 0;
+`
+
+const StepListContainer = styled.ol`
+  background-color: ${Colors.LIGHT_GRAY5};
+  display: flex;
+  align-items: center;
+  padding: 25px;
+  margin: 0;
+  border-radius: 3px 3px 0 0;
+`
+
+const StepListItemContainer = styled.li`
+  display: flex;
+  align-items: center;
+`
+
+type StepState = 'incomplete' | 'current' | 'complete'
+
+const StepListItemCircle = styled.div<{
+  state: StepState
+}>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 30px;
+  width: 30px;
+  border-radius: 50%;
+  opacity: ${props => (props.state === 'complete' ? 0.7 : 1)};
+  background-color: ${props =>
+    props.state === 'incomplete' ? Colors.GRAY4 : Colors.BLUE3};
+  margin-right: 10px;
+  color: ${Colors.WHITE};
+  font-weight: 500;
+`
+
+// eslint-disable-next-line no-unused-vars
+export const StepListItem = styled(({ current, ...props }) => (
+  <H5 {...props} />
+))<{ current: boolean }>`
+  color: ${props => (props.current ? Colors.DARK_GRAY5 : Colors.GRAY3)};
+  margin: 0;
+`
+
+const StepListItemLine = styled.div`
+  flex-grow: 1;
+  height: 1px;
+  background: ${Colors.GRAY5};
+  margin: 0 10px;
+`
+
+export const StepList: React.FC = ({ children }) => {
+  const stepItems = React.Children.toArray(children)
```

Comment:
I had two motivations:
- I'm going to build a step list where the step items are clickable and navigate back to the previous step. I figured it might be nice to be able to do that via composition, e.g. `<StepListItem><Link to='/login'>Log in</Link></StepListItem>`. Though who knows, it might just be easier to add an `onClick`
- I wanted to learn how to build composable components and see if I actually like this approach, because I've heard about it a lot.

Currently not sure how much I like it. Was planning to try it out for a bit and see how it felt. What do you think?

Reply:
Makes sense! Very open to continuing to try this path - easy enough to change down the road